### PR TITLE
- Fix for a route that has comments above it (see test case)

### PR DESCRIPTION
**Pull request by natefrechette**
- the fix will filter out any unnecessary comments to preserve the swagger docs between markers

### CODE REVIEW

File: src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala [hunk 1]
```diff
@@ -156,11 +159,17 @@ case class SwaggerSpecGenerator(
...
 
-      val commentDocLines = commentLines match {
+      @tailrec
+      def getCommentLines(commentLines: List[String]): List[String] = commentLines match {
         case `marker` +: docs :+ `marker` ⇒ docs
-        case _                            ⇒ Nil
+        case l: List[String] if l.nonEmpty ⇒
+          val sliced = commentLines.slice(commentLines.indexOf(marker), commentLines.lastIndexOf(marker) + 1)
+          getCommentLines(sliced)
+        case _ ⇒ Nil
```
Comment (kailuowang) [inline src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala]: In this implementation, the markers can be anywhere in the comments. I agree that requiring that the comment starts with a marker can be problematic when you want to comment out the previous doc. This implementation however might include docs that are in the above end point. 
I suggest that you change the implementation to enforce that the last comment line is the marker and only extract the lines between the last marker (also last line) and the second last marker. 
You don't have to use the pattern match to achieve that. I guess you can first test if the last line is a marker, and strip it and then takeRight with the right number of lines (could use `lastIndexOf(marker)` to help calculate. Need to also guard against the case when there isnt a second marker in the comments, in which case no docs.

**Comment by kailuowang** [comment]
Thanks for contributing again, @natefrechette. Really appreciate it.

File: src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala (hunk 1 above)
Comment (natefrechette) [inline src/main/scala/com/iheart/playSwagger/SwaggerSpecGenerator.scala]: I will check it out some more

**Comment by kailuowang** [comment]
@natefrechette btw, I have a question, are you using java 8 or 7? I know you are on play 2.3 right?

**Comment by natefrechette** [comment]
I am using java 8 & play 2.4.4 actually.  Locally I change the play version to 2.4.4 to get it to work

**Comment by kailuowang** [comment]
@natefrechette any plan to continue on this PR?

**Comment by kailuowang** [comment]
Close for now.

**Pull request closed without merging**